Makefile

```diff
@@ -0,0 +1,32 @@
+RANDOM_PORT := $(shell expr $$(( 8000 + (`id -u` % 1000) )))
+
+HUGO_VERSION ?= 0.54.0
+HUGO_URL := https://github.com/gohugoio/hugo/releases/download/v$(HUGO_VERSION)/hugo_$(HUGO_VERSION)_Linux-64bit.tar.gz
+HUGO_EXEC := vendor/hugo_$(HUGO_VERSION)
+
+.PHONY: all
+all: build
+
+.PHONY: build
+build: $(HUGO_EXEC)
+	$(HUGO_EXEC)
+
+.PHONY: dev
+dev: $(HUGO_EXEC)
+	$(HUGO_EXEC) --port $(RANDOM_PORT) server
+
+.PHONY: deploy
+deploy: build
+	git -C public fetch
+	git -C public reset --soft origin/gh-pages
+	git -C public add .
+	git -C public commit -m "Makefile deploy: $(USER)"
+	git -C public push origin 'HEAD:gh-pages'
```

Comment:
Is there a reason why you aren't doing this based on the instructions [in the Hugo docs](https://gohugo.io/hosting-and-deployment/hosting-on-github/#put-it-into-a-script-1)?

Reply:
Didn't know these docs existed lol. I'll take a look.

Reply:
Looks like they do something very similar but they use worktrees instead which is very nice actually. I probably will try and work it into the Makefile instead of writing an external script.